internal/transports/p2p/peer/utils.go

```diff
@@ -0,0 +1,44 @@
+package peer
+
+import (
+	"github.com/bitcoin-sv/block-headers-service/internal/chaincfg"
+	"github.com/bitcoin-sv/block-headers-service/internal/wire"
+)
+
+func minUint32(a, b uint32) uint32 {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+func findNextHeaderCheckpoint(checkpoints []chaincfg.Checkpoint, height int32) *chaincfg.Checkpoint {
+	if len(checkpoints) == 0 {
+		return nil
+	}
+
+	finalCheckpoint := &checkpoints[len(checkpoints)-1]
+	if height >= finalCheckpoint.Height {
+		return nil
+	}
+
+	nextCheckpoint := finalCheckpoint
+	for i := len(checkpoints) - 2; i >= 0; i-- {
+		if height >= checkpoints[i].Height {
+			break
+		}
+		nextCheckpoint = &checkpoints[i]
+	}
+	return nextCheckpoint
+}
+
+func searchForFinalBlock(invVects []*wire.InvVect) int {
```

Comment:
Consider add the name `searchForFinalBlockIndex` with `Index` suffix which describes what is returned.

Reply:
You're right, done